lib/api/2.0/notification.js

```diff
@@ -30,7 +36,124 @@ var notificationProgressPost = controller(function(req, res){
+            var parse = parser.urlencoded({ extended: false });
+            parse(req, res, function (err) {
+                if (err) {
+                    return reject(err);
+                }
+                    resolve();
+            });
+        })
+            .then(function() {
+            return waterline.catalogs.findOne({source: "bmc", "data.IP Address": ip})
+                .then(function(data) {
+                    if (data !== undefined && data.hasOwnProperty("node") && data.node !== ""){
+                         bmcCatalog = data;
+                        //Format the amqp message by adopting the rackHD notification standard
+                        amqpMessage.type = "node";
+                        amqpMessage.action = "alerts";
+                        amqpMessage.typeId = bmcCatalog.node;
+
+                        var httpReq = _.defaults(req.swagger.query || {},
+                            req.query || {},
+                            req.body || {});
+                        if(Object.keys(httpReq).length === 1 ) {
+                            var key = Object.keys(httpReq)[0];
+                            alert = JSON.parse(key);
+                            amqpMessage.data = alert;
+                            amqpMessage.data.sourceIpAddress = ip;
+                            return waterline.catalogs.findOne({source: "ipmi-fru"})
+                                .then(function(catalogIpmiFru){
+                                    if(catalogIpmiFru !== undefined &&
+                                        catalogIpmiFru.data
+                                            .hasOwnProperty("Builtin FRU Device (ID 0)")){
+                                        deviceInfoCatalog =
+                                            catalogIpmiFru.data["Builtin FRU Device (ID 0)"];
+
+                                        //Add useful information to the alert message
+                                        amqpMessage.data.rackhdId = bmcCatalog.node;
+                                        amqpMessage.data.macAddress = bmcCatalog["MAC Address"];
+                                        amqpMessage.severity = alert.Severity.toLowerCase();
+                                        amqpMessage.data.ChassisName =
+                                            deviceInfoCatalog["Board Product"];
+                                        amqpMessage.data.ServiceTag =
+                                            deviceInfoCatalog["Product Serial"];
+                                        amqpMessage.data.SN = deviceInfoCatalog["Board Serial"];
+
+                                        return notificationApiService.
+                                        alertNotification(amqpMessage);
+                                    }else{
+                                        //publish with the additional information
+                                        return notificationApiService.
```

Comment:
return notificationApiService.alertNotification(amqpMessage) is duplicated, you should remove the else block and end if block before this sentence instead.